### DISCUSSION TITLE
Release 5.2.0: Post installation  user creation is failing

### DISCUSSION THREAD

**Discussion by gohilamariappan** [Sunbird-Ed/Community#556] [Bugs]
HI Team!

while running the post install script user creation is giving 200 response with error message as "User is created but password couldn't be updated".
Tried the same in Postman apis. below one is the response.

{
    "id": "api.user.create",
    "ver": "v1",
    "ts": "2023-06-21 10:33:25:389+0000",
    "params": {
        "resmsgid": "",
        "msgid": "",
        "err": null,
        "status": "SUCCESS",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "userId": "",
        "response": "SUCCESS",
        "err_msg": "User is created but password couldn't be updated.",
        "errors": []
    }
}

Note: followed the steps which are mentioned in the below discussion
 https://github.com/orgs/Sunbird-Ed/discussions/306 

Below are the logs from learner service
############
d23-bfb7-77da32be5939\",\"type\":\"User\"},\"context\":{\"channel\":\"0138193046778920963\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.1.0\"},\"env\":\"User\",\"cdata\":[{\"id\":\"01337a5c50667630be7d339dda0cd46c\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"err\":\"UOS_0069\",\"stacktrace\":\"updatePassword: Exception occurred:  org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.handleErrorStatus(ClientInvocation.java:214)org.jboss.resteasy.client.jaxrs.internal.proxy.extractors.DefaultEntityExtractorFactory$3.extractEntity(DefaultEntityExtractorFactory.java:50)org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:107)org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:76)com.sun.proxy.$Proxy50.resetPassword(Unknown Source)org.sunbird.sso.impl.KeyCloakServiceImpl.updatePassword(KeyCloakServiceImpl.java:75)org.sunbird.util.user.UserUtil.updatePassword(UserUtil.java:298)org.sunbird.service.user.impl.SSOUserServiceImpl.createUserAndPassword(SSOUserServiceImpl.java:99)org.sunbird.actor.user.SSOUserCreateActor.processSSOUser(SSOUserCreateActor.java:117)org.sunbird.actor.user.SSOUserCreateActor.createSSOUser(SSOUserCreateActor.java:89)org.sunbird.actor.user.SSOUserCreateActor.onReceive(SSOUserCreateActor.java:61)org.sunbird.actor.core.BaseActor.onReceive(BaseActor.java:21)akka.actor.UntypedAbstractActor$$anonfun$receive$1.applyOrElse(AbstractActor.scala:291)akka.actor.Actor$class.aroundReceive(Actor.scala:517)akka.actor.AbstractActor.aroundReceive(AbstractActor.scala:180)akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)akka.actor.ActorCell.invoke(ActorCell.scala:557)akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)akka.dispatch.Mailbox.run(Mailbox.scala:225)akka.dispatch.Mailbox.exec(Mailbox.scala:235)akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\",\"errtype\":\"api_access\",\"requestid\":\"01337a5c50667630be7d339dda0cd46c\"}}","lname":"TelemetryEventLogger","tname":"application-akka.actor.most-used-one-dispatcher-149","level":"INFO","HOSTNAME":"learner-78c9854b7-j2jjk","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2023-06-21T12:23:09.187Z","msg":"{\"eid\":\"LOG\",\"ets\":1687350189187,\"ver\":\"3.0\",\"mid\":\"01337a5c50667630be7d339dda0cd46c\",\"actor\":{\"id\":\"9d536456-f23a-4d23-bfb7-77da32be5939\",\"type\":\"User\"},\"context\":{\"channel\":\"0138193046778920963\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.1.0\"},\"env\":\"User\",\"cdata\":[{\"id\":\"01337a5c50667630be7d339dda0cd46c\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"level\":\"info\",\"type\":\"Api_access\",\"message\":\"Cassandra query : [INSERT INTO sunbird.user_roles (role,createdDate,createdBy,scope,userId) VALUES ('ORG_ADMIN','2023-06-21 12:23:09:185+0000',null,'[{\\\"organisationId\\\":\\\"0138193046778920963\\\"}]','23adb2ab-89ea-4664-a57a-d19a9a49386c');]\",\"params\":[]}}","lname":"TelemetryEventLogger","tname":"application-akka.actor.most-used-one-dispatcher-149","level":"INFO","HOSTNAME":"learner-78c9854b7-j2jjk","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2023-06-21T12:23:09.187Z","msg":"{\"eid\":\"LOG\",\"ets\":1687350189187,\"ver\":\"3.0\",\"mid\":\"01337a5c50667630be7d339dda0cd46c\",\"actor\":{\"id\":\"9d536456-f23a-4d23-bfb7-77da32be5939\",\"type\":\"User\"},\"context\":{\"channel\":\"0138193046778920963\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.1.0\"},\"env\":\"User\",\"cdata\":[{\"id\":\"01337a5c50667630be7d339dda0cd46c\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"level\":\"info\",\"type\":\"Api_access\",\"message\":\"saveUserAttributes\",\"params\":[]}}","lname":"TelemetryEventLogger","tname":"application-akka.actor.most-used-one-dispatcher-149","level":"INFO","HOSTNAME":"learner-78c9854b7-j2jjk","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
{"timestamp":"2023-06-21T12:23:09.191Z","msg":"{\"eid\":\"LOG\",\"ets\":1687350189191,\"ver\":\"3.0\",\"mid\":\"01337a5c50667630be7d339dda0cd46c\",\"actor\":{\"id\":\"9d536456-f23a-4d23-bfb7-77da32be5939\",\"type\":\"User\"},\"context\":{\"channel\":\"0138193046778920963\",\"pdata\":{\"id\":\"dev.sunbird.learning.service\",\"pid\":\"learner-service\",\"ver\":\"5.1.0\"},\"env\":\"User\",\"cdata\":[{\"id\":\"01337a5c50667630be7d339dda0cd46c\",\"type\":\"Request\"}],\"rollup\":{}},\"edata\":{\"level\":\"info\",\"type\":\"Api_access\",\"message\":\"Cassandra query : INSERT INTO sunbird.user_organisation(hashTagId,isDeleted,organisationId,orgJoinDate,userId,associationType,id) VALUES (?,?,?,?,?,?,?);\",\"params\":[]}}","lname":"TelemetryEventLogger","tname":"application-akka.actor.rr-usr-dispatcher-124","level":"INFO","HOSTNAME":"learner-78c9854b7-j2jjk","application.home":"/home/sunbird/learner/learning-service-1.0-SNAPSHOT"}
##############################

Please suggest.

Regards,
Gohila

**Comment by chitranshu-keshav** [comment]
@rajeevsathish Can you please get back on this

**Comment by gohilamariappan** [comment]
Hi Team!

We are facing this issue due to empty value in the cassandra federation id (in postgres), we removed that and created a new federation id from the keycloak. Then it worked.

Regards,
Gohila
